src/mainboard/starfive/visionfive2/bt0/src/main.rs

```diff
@@ -127,11 +132,17 @@ pub unsafe extern "C" fn reset() {
         static _sidata: u8;
     }
 
-    let count = &_ebss as *const u8 as usize - &_sbss as *const u8 as usize;
-    ptr::write_bytes(&mut _sbss as *mut u8, 0, count);
-
-    let count = &_edata as *const u8 as usize - &_sdata as *const u8 as usize;
-    ptr::copy_nonoverlapping(&_sidata as *const u8, &mut _sdata as *mut u8, count);
+    let count =
+        addr_of_mut!(_ebss) as *const u8 as usize - addr_of_mut!(_sbss) as *const u8 as usize;
+    ptr::write_bytes(addr_of_mut!(_sbss) as *mut u8, 0, count);
+
+    let count =
+        addr_of_mut!(_edata) as *const u8 as usize - addr_of_mut!(_sdata) as *const u8 as usize;
```

Comment:
```suggestion
        addr_of!(_edata) as usize - addr_of!(_sdata) as usize;
```

Reply:
That is so cool, Rust is really getting more and more elegant all the time :) thank you!